waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -964,6 +973,25 @@
       has-reverse-proxy?
       (attach-envoy-sidecar reverse-proxy service-description base-env service-port port0))))
 
+(defn default-pdb-spec-builder
+  "Factory function which creates a Kubernetes PodDisruptionBudget spec for the given ReplicaSet."
+  [pdb-api-version rs-spec replicaset-uid]
+  (let [rs-api-version (get rs-spec :apiVersion)
+        rs-kind (get rs-spec :kind)
+        rs-name (get-in rs-spec [:metadata :name])
+        rs-selector (get-in rs-spec [:spec :selector])]
+    {:apiVersion pdb-api-version
+     :kind "PodDisruptionBudget"
+     :metadata {:name (str rs-name "-pdb")
+                :ownerReferences [{:apiVersion rs-api-version
+                                   :blockOwnerDeletion true
+                                   :controller false
+                                   :kind rs-kind
+                                   :name rs-name
+                                   :uid replicaset-uid}]}
+     :spec {:minAvailable 1
```

Comment:
Why min-available=1 rather than max-unavailable=1?

Reply:
I chose `minAvailable=1` as that is least disruptive for cluster operators. 

Changed it to `(quot (inc min-instances) 2)` now.
This formula works even if the function is called with `min-instances=1`.